Services/HubSpotService.cs

```diff
@@ -30,4 +31,49 @@ public async void CreateContact(/* string ownerName, */ string emailAddress, /*
             // log here
         }
     }
+
+    public async void UpdateContact(string? name, string emailAddress, UserStatus? userStatus, string? phoneNumber)
+    {
+        _client.DefaultRequestHeaders.Add("Authorization", "Bearer " + configuration["HUBSPOT_API_KEY"]);
+
+        var properties = new Dictionary<string, object>
+        {
+            { "email", emailAddress }
+        };
+
+        if (!string.IsNullOrEmpty(name))
+        {
+            properties.Add("firstname", name);
+        }
+
+        if (userStatus.HasValue)
+        {
+            properties.Add("user_status", userStatus.ToString()!);
+        }
+
+        if (!string.IsNullOrEmpty(phoneNumber))
+        {
+            properties.Add("phone", phoneNumber);
+        }
+
+        var jsonBody = new
+        {
+            properties = properties
+        };
+
+        var content = new StringContent(
+            System.Text.Json.JsonSerializer.Serialize(jsonBody),
+            Encoding.UTF8,
+            "application/json"
+        );
+
+        var response = await _client.PatchAsync($"https://api.hubapi.com/crm/v3/objects/contacts/{emailAddress}?idProperty=email", content);
```

Comment:
How does this call authenticate with hubSpot?

Reply:
we set the auth header value to "bearer" our_api_key